tests/profiling/test_profiler.py

```diff
@@ -79,6 +79,33 @@ def test_tracer_api(monkeypatch):
         pytest.fail("Unable to find stack collector")
 
 
+def test_profiler_init_float_division_regression(run_python_code_in_subprocess):
+    """
+    Regression test for https://github.com/DataDog/dd-trace-py/pull/3751
+      When float division is enabled, the value of `max_events` can be a `float`,
+        this is then passed as `deque(maxlen=float)` which is a type error
+
+    File "/var/task/ddtrace/profiling/recorder.py", line 80, in _get_deque_for_event_type
+    return collections.deque(maxlen=self.max_events.get(event_type, self.default_max_events))
+    TypeError: an integer is required
+    """
+    code = """
+from ddtrace.profiling import profiler
+from ddtrace.profiling.collector import stack_event
+
+prof = profiler.Profiler()
+
+# The error only happened for this specific kind of event
+# DEV: Yes, this is likely a brittle way to test, but quickest/easiest way to trigger the error
+prof._recorder.push_event(stack_event.StackExceptionSampleEvent())
```

Comment:
if this doesn't fail on every version of Python we could add `from __future__ import division` to ensure float division is used.

Or maybe it doesn't matter since if it doesn't happen by default on 2.7 then what difference does it make?


```
# tox --skip-pkg-install -e 'py39-profile' -- -k regression
py39-profile installed: attrs==21.4.0,bytecode==0.13.0,cmake==3.22.4,coverage==6.4,Cython==0.29.30,ddsketch==2.0.2,-e git+git@github.com:DataDog/dd-trace-py.git@bd4c0d515b1d526a664ff5b2eaa559770f8f2467#egg=ddtrace,hypothesis==6.46.7,iniconfig==1.1.1,mock==4.0.3,ninja==1.10.2.3,opentracing==2.4.0,packaging==21.3,pluggy==1.0.0,protobuf==3.20.1,py==1.11.0,py-cpuinfo==8.0.0,pyparsing==3.0.9,pytest==7.1.2,pytest-asyncio==0.18.3,pytest-benchmark==3.4.1,pytest-cov==3.0.0,pytest-mock==3.7.0,six==1.16.0,sortedcontainers==2.4.0,tenacity==8.0.1,tomli==2.0.1,uWSGI==2.0.20
py39-profile run-test-pre: PYTHONHASHSEED='3332301474'
py39-profile run-test: commands[0] | python -m tests.profiling.run pytest --no-cov --capture=no --verbose --benchmark-disable -k regression tests/profiling
================================================================================================== test session starts ==================================================================================================
platform linux -- Python 3.9.11, pytest-7.1.2, pluggy-1.0.0 -- /root/project/.tox/py39-profile/bin/python
cachedir: .tox/py39-profile/.pytest_cache
hypothesis profile 'default' -> suppress_health_check=[HealthCheck.too_slow], database=DirectoryBasedExampleDatabase('/root/project/.hypothesis/examples')
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /root/project, configfile: setup.cfg
plugins: ddtrace-1.1.0rc2.dev77+gbd4c0d515.d20220523, hypothesis-6.46.7, asyncio-0.18.3, benchmark-3.4.1, cov-3.0.0, mock-3.7.0
asyncio: mode=legacy
collected 161 items / 160 deselected / 1 selected

tests/profiling/test_profiler.py::test_profiler_init_float_division_regression FAILED

======================================================================================================= FAILURES ========================================================================================================
_____________________________________________________________________________________ test_profiler_init_float_division_regression ______________________________________________________________________________________

run_python_code_in_subprocess = <function run_python_code_in_subprocess.<locals>._run at 0x7f72887af550>

    def test_profiler_init_float_division_regression(run_python_code_in_subprocess):
        """
        Regression test for https://github.com/DataDog/dd-trace-py/pull/3751
          When float division is enabled, the value of `max_events` can be a `float`,
            this is then passed as `deque(maxlen=float)` which is a type error

        File "/var/task/ddtrace/profiling/recorder.py", line 80, in _get_deque_for_event_type
        return collections.deque(maxlen=self.max_events.get(event_type, self.default_max_events))
        TypeError: an integer is required
        """
        code = """
    from ddtrace.profiling import profiler
    from ddtrace.profiling.collector import stack_event

    prof = profiler.Profiler()

    # The error only happened for this specific kind of event
    prof._recorder.push_event(stack_event.StackExceptionSampleEvent())
        """

        out, err, status, _ = run_python_code_in_subprocess(code)
>       assert status == 0, err
E       AssertionError: b'Traceback (most recent call last):
E           File "/tmp/pytest-of-root/pytest-9/test_profiler_init_float_divis0/test.py", l...ollections.deque(maxlen=self.max_events.get(event_type, self.default_max_events))
E         TypeError: an integer is required
E         '
E       assert 1 == 0

tests/profiling/test_profiler.py:103: AssertionError
```